includes/admin/class-omise-page-card-form-customization.php

```diff
@@ -0,0 +1,130 @@
+
+if (class_exists('Omise_Page_Settings')) {
+	return;
+}
+
+class Omise_Page_Card_From_Customization extends Omise_Admin_Page
+{
+
+	const PAGE_NAME = 'omise_card_form_customization_option';
+
+	protected function get_default_design_setting()
+	{
+		$theme = (new Omise_Payment_Creditcard())->get_option('card_form_theme');;
+		if(empty($theme)) {
+			$theme = 'light';
+		}
+		$design = [
+			'light' => [
+				'font' => [
+					'name' => 'Poppins',
+					'size' => 16,
+				],
+				'input' => [
+					'height' => '44px',
+					'border_radius' => '4px',
+					'border_color' => '#ced3de',
+					'active_border_color' => '#1451cc',
+					'background_color' => '#ffffff',
+					'label_color' => '#212121',
+					'text_color' => '#212121',
+					'placeholder_color' => '#98a1b2',
+				],
+				'checkbox' => [
+					'text_color' => '#1c2433',
+					'theme_color' => '#1451cc',
+				]
+			],
+			'dark' => [
+				'font' => [
+					'name' => 'Poppins',
+					'size' => 16,
+				],
+				'input' => [
+					'height' => '44px',
+					'border_radius' => '4px',
+					'border_color' => '#475266',
+					'active_border_color' => '#475266',
+					'background_color' => '#131926',
+					'label_color' => '#E6EAF2',
+					'text_color' => '#212121',
+					'placeholder_color' => '#DBDBDB',
+				],
+				'checkbox' => [
+					'text_color' => '#E6EAF2',
+					'theme_color' => '#1451CC',
+				]
+			]
+		];
+		return $design[$theme] ?? $design['light'];
+	}
+
+	/**
+	 * get design setting
+	 */
+	public function get_design_setting()
+	{
+		$formDesign = get_option(self::PAGE_NAME);
+		if (empty($formDesign)) {
+			$formDesign = $this->get_default_design_setting();
+		}
+		return $formDesign;
+	}
+
+	/**
+	 * @param array $data
+	 *
+	 * @since  3.1
+	 */
+	protected function save($data)
+	{
+		if (!isset($data['omise_setting_page_nonce']) || !wp_verify_nonce($data['omise_setting_page_nonce'], 'omise-setting')) {
+			wp_die(__('You are not allowed to modify the settings from a suspicious source.', 'omise'));
+		}
+		$options = [];
+		$defaultValues = $this->get_default_design_setting();
+		foreach ($defaultValues as $componentKey => $componentValue) {
+			foreach ($componentValue as $key => $val) {
+				$options[$componentKey][$key] = sanitize_text_field($data[$componentKey][$key]);
+			}
+		}
+		update_option(self::PAGE_NAME, $options);
+		$this->add_message('message', "Update has been saved!");
+	}
+
+	/**
+	 * @param array $data
+	 *
+	 * @since  3.1
+	 */
+	protected function reset_default_setting($data)
+	{
+		if (!isset($data['omise_setting_page_nonce']) || !wp_verify_nonce($data['omise_setting_page_nonce'], 'omise-setting')) {
+			wp_die(__('You are not allowed to modify the settings from a suspicious source.', 'omise'));
+		}
+		update_option(self::PAGE_NAME, null);
+		$this->add_message(
+			'message',
+			"Setting have been reset!"
+		);
+	}
+
+	/**
+	 * @since  3.1
+	 */
+	public static function render()
+	{
+		$page = new self;
```

Comment:
Let's consider using the Singleton pattern here to ensure that only a single instance of the class can be instantiated.